src/libs/ReportUtils.ts

```diff
@@ -7999,19 +8001,18 @@ function parseReportRouteParams(route: string): ReportRouteParams {
         return {reportID: '', isSubReportPageRoute: false};
     }
 
-    const pathSegments = parsingRoute.split('/');
+    const state = getStateFromPath(parsingRoute as Route);
+    const focusedRoute = findFocusedRoute(state);
 
-    const reportIDSegment = pathSegments.at(1);
-    const hasRouteReportActionID = !Number.isNaN(Number(reportIDSegment));
+    const reportID = focusedRoute?.params && 'reportID' in focusedRoute.params ? (focusedRoute?.params?.reportID as string) : '';
 
-    // Check for "undefined" or any other unwanted string values
-    if (!reportIDSegment || reportIDSegment === 'undefined') {
+    if (!reportID) {
         return {reportID: '', isSubReportPageRoute: false};
     }
 
     return {
-        reportID: reportIDSegment,
-        isSubReportPageRoute: pathSegments.length > 2 && !hasRouteReportActionID,
+        reportID,
+        isSubReportPageRoute: focusedRoute?.name !== SCREENS.REPORT,
     };
```

Comment:
I am wondering if maybe we could just return a focusedRoute if it is a report route?

1. `reportID` is not the only param of the report screen which could be missleading
2. `isSubReportPageRoute` for binary situation. What if we get more types of report screens? IMO it's more readable if we just check the name of the route in place where we want to use parseReportRouteParams. 

As an afterthought, maybe we could change this function to just `parseRoute`?

Then check the name in place where this function is used. It looks like it is used only in two places.

On of these places is `getReportIDFromLink`. It look a little redundant to me.

Reply:
If it's only for the `SCREENS.REPORT`, how is this line supposed to work? 
```
isSubReportPageRoute: focusedRoute?.name !== SCREENS.REPORT,
```
It will always be `false`

Reply:
Ok makes sense. Could you please add comment explaining what sub report route is? I don't think this term is widely used in the app.